File: src/types/policy.ts
// SPDX-FileCopyrightText: 2026 Pascal Kuhn <support@oxidvault.com>
// SPDX-License-Identifier: AGPL-3.0-only

export interface ResolvedBoolField {
  value: boolean;
  disabled: boolean;
}

export interface ResolvedU32Field {
  value: number;
  disabled: boolean;
}

export interface ResolvedConfig {
  adminPolicyActive: boolean;
  forceLockOnMinimize: ResolvedBoolField;
  autoLockSeconds: ResolvedU32Field;
  gitSyncEnabled: ResolvedBoolField;
  minMasterPasswordLen: ResolvedU32Field;
}

File: src-tauri/src/commands/policy.rs
// SPDX-FileCopyrightText: 2026 Pascal Kuhn <support@oxidvault.de>
// SPDX-License-Identifier: AGPL-3.0-only

use tauri::AppHandle;
use vault_core::policy::{resolve_config, ResolvedConfig};

use crate::settings::load_settings;

#[tauri::command]
pub fn get_resolved_config(app: AppHandle) -> Result<ResolvedConfig, String> {
    let settings = load_settings(&app)?;
    Ok(resolve_config(&settings.policy_preferences()))
}
